### ISSUE TITLE
ES2015 export does not work in plugin development

### ISSUE THREAD

**Issue by screendriver** [systemjs/systemjs#774]
Once again I have trouble on developing my plugin. But this time I believe it's a bug in the bundling system.

I created a [Gist](https://gist.github.com/screendriver/1bbba8079421fa520128). The bug occurs at [line 19](https://gist.github.com/screendriver/1bbba8079421fa520128#file-sass-js-L19) in `sass.js`. If you run the script it displays `foo` at the command line. This works as expected because I use CommonJS syntax to export `foo` (see [sass-builder.js](https://gist.github.com/screendriver/1bbba8079421fa520128#file-sass-builder-js)). The bug however occurs when you import [sass-builder-es2015.js](https://gist.github.com/screendriver/1bbba8079421fa520128#file-sass-builder-es2015-js) instead of [sass-builder.js](https://gist.github.com/screendriver/1bbba8079421fa520128#file-sass-builder-js) at [line 17](https://gist.github.com/screendriver/1bbba8079421fa520128#file-sass-js-L17). The command line output displays now `{}` instead of the default export. So it is not possible to write ES2015 code when you use `loader.import('./some-file', { name: __moduleName })`

**Comment by guybedford** [comment]
@screendriver you'd need to manually read `builder.default` for the sass-builder-es2015 case. If this isn't working though then it sounds like a SystemJS Builder issue. I've actually been working on this mechanic for the coming release so it may already been fixed on master if so.

**Comment by screendriver** [comment]
Unfortunately it isn't working.
Should we leave this issue open until I make a retest in the next release?

**Comment by guybedford** [comment]
Sure, that would be great, thanks.

**Comment by guybedford** [comment]
This should be working in the latest release now.

**Comment by screendriver** [comment]
Unfortunately not. Nothing changed.

**Comment by guybedford** [comment]
It definitely would have fixed it - are you sure you are running 0.16.4?
Try 'jspm -v' to confirm.

**Comment by screendriver** [comment]
```sh
❯ jspm -v
0.16.4
```

You can simply reproduce it with:

```js
let fetch;
let translate;
let bundle;

if (typeof window !== 'undefined') {
  // Browser runtime code
} else {
  // setting format = 'defined' means we're managing our own output
  translate = load => {
    load.metadata.format = 'defined';
  };
  bundle = function bundler(loads, opts) {
    return System.import('./sass-builder', { name: __moduleName })
      .then(builder => {
        console.log(builder);
        builder.call(System, loads, opts)
      });
  };
}

export {fetch, translate, bundle};
```

and my `sass-builder.js`

```js
export const foo = 'bar';
```

The `console.log()` statement above prints `{}` instead of my exported `foo` variable. If I change `sass-builder.js` to

```js
exports.foo = 'bar';
```

`console.log()` prints `{ foo: 'bar' }`

**Comment by guybedford** [comment]
@screendriver did you run `jspm dl-loader` here? In my simple test I'm definitely getting a `builder.foo` as defined here during `jspm bundle`.

**Comment by screendriver** [comment]
Yes I did that. I've pushed a test branch to my repository so you can easily reproduce it. Just checkout https://github.com/screendriver/plugin-sass.git , switch to the newly created `es2015` branch and do a

```sh
$ npm install
...
$ jspm install
...
$ gulp test:bundle
```

At the command line you see now a test console statement `Foo? {}` instead of `Foo? { foo: 'bar' }`.

**Comment by guybedford** [comment]
@screendriver ES modules export always a namespace object to System.import, so in https://github.com/screendriver/plugin-sass/blob/es2015/src/sass.js#L18 you need to access `builder.foo`.

**Comment by screendriver** [comment]
@guybedford but excatly this is this the issue. `builder.foo` is undefined because the `builder` object is only `{}` instead of `{ foo: 'bar' }`.
If I write CommonJS like `exports.foo = 'bar';` instead of ES2015 modules `export const foo = 'bar';` it's working.
Just simply check out my `es2015` branch in my [repo](https://github.com/screendriver/plugin-sass.git) and you can see what I mean.

**Comment by guybedford** [comment]
I did run on the demo on that branch and `builder.foo` was logging `bar` fine.

**Comment by screendriver** [comment]
Hmm do have a clue what's going wrong here at my machine? I've tried it on two different machines and both have the same issue.

**Comment by guybedford** [comment]
If `jspm -v` is at latest, and you've run `jspm dl-loader` I'd be very surprised if `console.log('Foo?', builder.foo);` in the plugin import wasn't returning the right value.

**Comment by screendriver** [comment]
But it is so :cry: 
Can you do me a favor? I created an archive which can be downloaded [here](hs.echooff.de/plugin_sass.tar.gz). Everything, expect the hidden `.git` directory`, is included and can be directly run from the command line. Can you test at your machine if it's working there? That would be great.

**Comment by guybedford** [comment]
Here's the output I get:

```
jspm bundle 'test/bundleme.js'
     Building the bundle tree for test/bundleme.js...
     
       test/bundleme.js
       test/test.scss!src/sass.js
     
Foo? bar

err  TypeError: undefined is not a function
         at file:///Users/guybedford/Downloads/plugin-sass/src/sass.js!transpiled:26:21
         at runMicrotasksCallback (node.js:337:7)
         at process._tickCallback (node.js:355:11)
Guys-MacBook-Air:plugin-sass guybedford$ sublime src/sass.j
```

**Comment by screendriver** [comment]
~~Hmm that's unbelievable :flushed: I don't understand that. How can this be? Maybe because I am using Arch Linux and iojs...~~

Ah! I found it! I see the difference. You are using `jspm bundle` from the command line and **not** the Gulp task that I have defined. There I use [SystemJS builder](https://github.com/systemjs/builder). If I use the command line tool it works as well, but if I use SystemJS builder it fails. So it seems that it's a :beetle: in SystemJS builder?
